custom_components/xiaomi_home/miot/miot_lan.py

```diff
@@ -954,175 +1013,137 @@ def __gen_msg_id(self) -> int:
-                self._mev.loop_stop()
+            _LOGGER.error('send2device error, %s', err)
+            handler({
+                'code': MIoTErrorCode.CODE_INTERNAL_ERROR.value,
+                'error': str(err)},
+                handler_ctx)
+
+    def __sub_device_state(self, data: _MIoTLanSubDeviceData) -> None:
+        self._device_state_sub_map[data.key] = data
+
+    def __unsub_device_state(self, data: _MIoTLanUnsubDeviceData) -> None:
+        self._device_state_sub_map.pop(data.key, None)
+
+    def __sub_broadcast(self, data: _MIoTLanRegisterBroadcastData) -> None:
+        self._device_msg_matcher[data.key] = data
+        _LOGGER.debug('lan register broadcast, %s', data.key)
+
+    def __unsub_broadcast(self, data: _MIoTLanUnregisterBroadcastData) -> None:
+        if self._device_msg_matcher.get(topic=data.key):
+            del self._device_msg_matcher[data.key]
+        _LOGGER.debug('lan unregister broadcast, %s', data.key)
+
+    def __get_dev_list(self, data: _MIoTLanGetDevListData) -> None:
+        dev_list = {
+            device.did: {
+                'online': device.online,
+                'push_available': device.subscribed
+            }
+            for device in self._lan_devices.values()
+            if device.online}
+        data.handler(
+            dev_list, data.handler_ctx)
+
+    def __update_devices(self, devices: dict[str, dict]) -> None:
+        for did, info in devices.items():
+            # did MUST be digit(UINT64)
+            if not did.isdigit():
+                _LOGGER.info('invalid did, %s', did)
+                continue
+            if (
+                    'model' not in info
+                    or info['model'] in self._profile_models):
+                # Do not support the local control of
+                # Profile device for the time being
+                _LOGGER.info(
+                    'model not support local ctrl, %s, %s',
+                    did, info.get('model'))
+                continue
+            if did not in self._lan_devices:
+                if 'token' not in info:
+                    _LOGGER.error(
+                        'token not found, %s, %s', did, info)
+                    continue
+                if len(info['token']) != 32:
+                    _LOGGER.error(
+                        'invalid device token, %s, %s', did, info)
+                    continue
+                self._lan_devices[did] = _MIoTLanDevice(
+                    manager=self, did=did, token=info['token'],
+                    ip=info.get('ip', None))
+            else:
+                self._lan_devices[did].update_info(info)
+
+    def __delete_devices(self, devices: list[str]) -> None:
+        for did in devices:
+            lan_device = self._lan_devices.pop(did, None)
+            if not lan_device:
+                continue
+            lan_device.on_delete()
+
+    def __on_network_info_chnage(self, data: _MIoTLanNetworkUpdateData) -> None:
```

Comment:
spelling mistakes

Reply:
Fixed in 92ff6a4cf11f0935c5a39b1c56790bc775900564